src/main/perl/Blockdevices.pm

```diff
@@ -17,16 +17,23 @@ use constant BLKID => "/sbin/blkid";
 use constant FILES => qw (file -s);
 
 use constant PART_FILE  => '/tmp/created_partitions';
-use constant HOSTNAME	=> "/system/network/hostname";
-use constant DOMAINNAME	=> "/system/network/domainname";
+use constant HOSTNAME   => "/system/network/hostname";
+use constant DOMAINNAME => "/system/network/domainname";
+use constant KS         => "/system/aii/osinstall/ks";
 
 use constant GET_SIZE_BYTES  => qw (/sbin/blockdev --getsize64);
 
+# Lowest supported version is 5.0
+use constant ANACONDA_VERSION_EL_5_0 => version->new("11.1");
+use constant ANACONDA_VERSION_EL_6_0 => version->new("13.21");
+use constant ANACONDA_VERSION_EL_7_0 => version->new("19.31");
+use constant ANACONDA_VERSION_LOWEST => ANACONDA_VERSION_EL_5_0;
+
 use parent qw(CAF::Object Exporter);
 
 our $reporter = $main::this_app;
 
-our @EXPORT_OK = qw ($reporter PART_FILE);
+our @EXPORT_OK = qw ($reporter PART_FILE ANACONDA_VERSION_EL_7_0);
```

Comment:
export the otehr constants and the get_anaconda_version function/method; so we can import this in ks.pm and pxelinux.pm (nistead of copying it even more)

Reply:
I feel that's the wrong direction - a library called "blockdevices" is not the right place for such methods.

I'd rather move in the direction of moving all the Kickstart formatting from `ncm-lib-blockdevices` to `aii-ks`. `ncm-lib-blockdevices` should just provide an iterator function, which gets a formatter class as argument; then `aii-ks` could provide two formatter classes, one for Kickstart stanzas, the other for `%pre` script fragments. I'm just not quite sure how to implement that in Perl - my Perl-fu is getting rusty :-)

Btw. such an iterator which implements topological ordering would be needed anyway, to fix some corner cases. Starting from filesystems, like `aii-ks` is doing from the moment, and then working backwards to create the dependencies, does not allow some constructs like creating an empty volume group with no LVs inside, or creating a partition table only if a disk is empty, but leaving existing content untouched otherwise.

Reply:
ok, that is indeed better. but that also sounds like a lot of work, not for this or the other aii PRs